### ISSUE TITLE
Accessibility: sorting functionality needs to be done via buttons

### ISSUE THREAD

**Issue by jechols** [Mottie/tablesorter#1688]
There are some accessibility problems with the sortable headers generated:

- An element defined as a row header doesn't have any activation semantics.  Having it act as both a button and a row header is a violation of WCAG 4.1.2 ("Name, role, value": https://www.w3.org/WAI/WCAG20/quickref/20160105/#ensure-compat-rsv), a level A requirement (A is the most important to meet).
- The header and the sort button must be separate elements - the header must exist to declare its semantics as a row header separately from a button which lets users know that there is action which can be taken.  (e.g., you can't have one element that defines two separate pieces of functionality)

If a semantic `<button>` isn't doable / desirable, an element must be built that functions as a button and declares itself to be a button via the "button" role.  It must respond to clicks, the "enter" key, and the space bar; it must be in the tab order; etc.  A button achieves all this without any custom JavaScript, but it's not required.  The W3C's reference implementation uses a span that's coded to act as a button: http://w3c.github.io/aria-practices/examples/grid/dataGrids.html#ex2_label.  (Their example is for a table coded as a data grid, so certain things may not apply directly to a simpler table, but W3C's examples tend to be excellent for reference compared to looking up stuff on a random blog.)

**Comment by Mottie** [comment]
Hi @jechols!

Thanks for sharing the link and information. You're right, I haven't been keeping this library up to date with the accessibility standards. I think updating would require some substantial internal changes which would warrant a major release. Sadly, I haven't found the time to complete my already planned rewrite. But I will be sure to save these links and follow the current standard.

**Comment by jechols** [comment]
Thanks for the response, I definitely understand the difficulties in remediation on codebases that didn't start out with accessibility features as a core principle.  I'll watch for your next update, and maybe poke around in the code, though if you think it's a huge change I doubt I can help :(
